internal/app/referenceclient/client.go

```diff
@@ -109,16 +114,39 @@ func invoke(ctx context.Context, req *v1alpha1.ClientCompatRequest) (*v1alpha1.C
 	var transport http.RoundTripper
 	switch req.HttpVersion {
 	case v1alpha1.HTTPVersion_HTTP_VERSION_1:
-		transport = &http.Transport{}
+		if tlsConf != nil {
+			tlsConf.NextProtos = []string{"http/1.1"}
```

Comment:
This forces the client to use HTTP 1.1 with TLS. Without this, since `net/http` supports both HTTP 1.1 _and_ HTTP/2, it would end up trying to negotiate HTTP/2. But the `NextProtos` slice is what defines the set of supported protocols. (Same goes for why we set this to `[]string{"h2"}` below, to make sure we don't unintentionally negotiate HTTP 1.1 with the server.)

Reply:
`NextProtos` is such an unfortunate name. I kept thinking 'why does TLS care about proto files'